### PR TITLE
Set noble as default ubuntu distribution.

### PR DESCRIPTION
**Pull request by marcoag**
Adding noble as default ubuntu distribution.

This shouldn't affect `iron` or `humble` as they seem to override this value:

https://github.com/ros2/ci/blob/272db1881d96e45b9476da316ca667ef59067af9/create_jenkins_job.py#L462-L463

https://github.com/ros2/ci/blob/272db1881d96e45b9476da316ca667ef59067af9/create_jenkins_job.py#L476-L477

### CODE REVIEW

**Comment by clalancette** [comment]
As it stands, the one reason we can't do this is because `python3-flake8` is still broken in Noble.  @j-rivero is working on getting that solved in Noble proper, and @claraberendsen may be working on a workaround (by pulling the package from noble-proposed) for the short-term.

**Comment by clalancette** [comment]
Before we can merge this one in, we need https://github.com/ros2/ci/pull/766 and some of the fixes from https://github.com/ros2/ci/pull/761 as well.

**Review by wjwwood — APPROVED** [review]

**Comment by wjwwood** [comment]
lgtm pending https://github.com/ros2/ci/pull/761

**Comment by clalancette** [comment]
All right, I believe that we have everything we need now.  Here is CI on Noble with the current state of the repository:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20782/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15078/)
* Clang [Build Status](http://ci.ros2.org/job/ci_linux_clang_libcxx/66/)

**Comment by clalancette** [comment]
Bah, we still need another PR.  See https://github.com/ros2/ci/pull/769

**Comment by clalancette** [comment]
All right, all CI for this was done in https://github.com/ros2/ci/pull/769 .  I'm going to approve, merge, and then deploy this change so we start using Noble as the default for CI.

**Review by clalancette — APPROVED** [review]

**Pull request merged**
